dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -324,6 +347,31 @@ public void Dispose()
         return result;
     }
 
+    /// <summary>
+    /// Execute the FunctionInvoking event handlers and returns true if no cancellation was attempted.
+    /// </summary>
+    /// <param name="functionView">Function view details</param>
+    /// <param name="context">SKContext before function invocation</param>
+    /// <param name="prompt">Generated prompt when using semantic functions</param>
+    /// <returns>Returns true if no cancellation was attempted.</returns>
+    private bool OnFunctionInvoking(FunctionView functionView, SKContext context, string? prompt)
+    {
+        var args = new FunctionInvokingEventArgs(functionView, context, prompt);
+        this.FunctionInvoking?.Invoke(this, args);
+
+        return !args.Cancel;
```

Comment:
Better to avoid allocating the arguments if there's no handler registered.
```suggestion
        var func = this.FunctionInvoking;
        if (func is not null)
        {
            var args = new FunctionInvokingEventArgs(functionView, context, prompt);
            func(this, args);
            return !args.Cancel;
        }
        
        return true;
```